activerecord/lib/active_record/model_schema.rb

```diff
@@ -499,6 +500,30 @@ def compute_table_name
             base.table_name
           end
         end
+
+        def warn_if_deprecated_type(column)
+          return if attributes_to_define_after_schema_loads.key?(column.name)
+          return unless column.respond_to?(:oid)
+          if column.array?
+            array_arguments = ", array: true"
+          else
+            array_arguments = ""
+          end
+          if column.sql_type.start_with?("interval")
```

Comment:
Yes, you will need to provide a deprecation warning. There's no reason to have a `legacy_interval` type, the user can just do `attribute :foo, :string`